--- a/ov_msckf/src/ros/ROS2Visualizer.cpp	2026-10-07 07:17:57.504143283 +1300
+++ b/ov_msckf/src/ros/ROS2Visualizer.cpp	2026-10-07 07:17:57.506692607 +1300
@@ -170,7 +170,7 @@
   _node->declare_parameter<std::string>("topic_imu", "/imu0");
   _node->get_parameter("topic_imu", topic_imu);
   parser->parse_external("relative_config_imu", "imu0", "rostopic", topic_imu);
-  sub_imu = _node->create_subscription<sensor_msgs::msg::Imu>(topic_imu, rclcpp::SensorDataQoS(),
+  sub_imu = _node->create_subscription<sensor_msgs::msg::Imu>(topic_imu, rclcpp::SensorDataQoS().keep_last(1000),
                                                               std::bind(&ROS2Visualizer::callback_inertial, this, std::placeholders::_1));
   PRINT_INFO("subscribing to IMU: %s\n", topic_imu.c_str());
 
